test/test_syntax.tt

```diff
@@ -1887,4 +1887,37 @@ now => @com.facebook.post();
 // $context.result
 $dialogue @org.thingpedia.dialogue.transaction.execute;
 @com.thecatapi.get();
-@com.twitter.post_picture(picture_url=$context.result.picture_url : Entity(tt:picture));
\ No newline at end of file
+@com.twitter.post_picture(picture_url=$context.result.picture_url : Entity(tt:picture));
+
+====
+
+// existential subquery
+// simple invocation no filter
+monitor(@com.twitter.home_timeline() filter any(@org.thingpedia.builtin.thingengine.builtin.get_gps()));
+
+
+====
+
+// existential subquery
+// projection
+monitor(@com.twitter.home_timeline() filter any([speed] of @org.thingpedia.builtin.thingengine.builtin.get_gps()));
```

Comment:
Should we optimize out projections inside existential subqueries? They have no effect, right?

Reply:
I think it's possible for a query to have an optional output parameter (such as get tweet may contain a picture or not). As we group functionalities based on natural language, we probably will keep this design. So projection here is meaningful. I think the compiler code is not reflecting this design though... so that needs to be updated...

Reply:
ok so the semantics of existential projection means "there exist at least one row for which all the projected values are not `null` or `undefined`" is that right?

Reply:
right